src/v0/destinations/lemnisk/transform.js

```diff
@@ -0,0 +1,136 @@
+const { EventType } = require("../../../constants");
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  DIAPI_CONFIG_CATEGORIES,
+  DIAPI_MAPPING_CONFIG
+} = require("./config");
+const {
+  constructPayload,
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+  defaultRequestConfig,
+  simpleProcessRouterDest
+} = require("../../util");
+const { generatePageName, fetchPlatform } = require("./utils");
+const {
+  ConfigurationError,
+  TransformationError,
+  InstrumentationError
+} = require("../../util/errorTypes");
+
+const responseBuilder = (message, category, destination, platform) => {
+  let payload;
+  const { Config } = destination;
+  const response = defaultRequestConfig();
+  response.method = defaultPostRequestConfig.requestMethod;
+  if (platform === "pl") {
+    const { plWriteKey, pl } = Config;
+    if (!plWriteKey || !plWriteKey.length > 0 || !pl || !pl.length > 0) {
+      throw new ConfigurationError(
+        "Configuration for Web Mode requires write key and region url"
+      );
+    }
+    payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+    if (category.type === "page") {
+      payload.name = generatePageName(message);
+    }
+    payload.writeKey = plWriteKey;
+    payload.context.userAgent = {
+      ua: payload.context.userAgent
+    };
+    response.endpoint = pl;
+    response.headers = {
+      "Content-Type": "application/json"
+    };
+  } else {
+    // diapi
+    const { diapi, apiKey, passKey, diapiWriteKey, srcId } = Config;
+    if (
+      !diapi ||
+      !diapi.length > 0 ||
+      !apiKey ||
+      !apiKey.length > 0 ||
+      !passKey ||
+      !passKey.length > 0
+    ) {
+      throw new ConfigurationError(
+        "Configuration for Server Mode requires Api key, Pass Key and region url"
+      );
+    }
+    payload = constructPayload(message, DIAPI_MAPPING_CONFIG[category.name]);
+    if (diapiWriteKey !== "") {
+      payload.WriteKey = diapiWriteKey;
+    }
+    if (srcId !== "") {
+      payload.srcid = srcId;
+    }
+    response.endpoint = diapi;
+    response.headers = {
+      "Content-Type": "application/json",
+      "x-api-passKey": passKey,
+      "x-api-key": apiKey
+    };
+  }
+  response.userId = message.anonymousId || message.userId;
+  if (payload) {
+    payload.type = category.type;
+    response.body.JSON = removeUndefinedAndNullValues(payload);
+  } else {
+    // fail-safety for developer error
+    throw new TransformationError("Payload could not be constructed");
+  }
+  return response;
+};
+
+const process = event => {
+  const { message, destination } = event;
+  if (!message.type) {
+    throw new InstrumentationError("Event type is required");
+  }
+
+  const messageType = message.type.toLowerCase();
+  let response;
+  let category;
+  const platform = fetchPlatform(destination);
+  if (platform === "pl") {
+    switch (messageType) {
+      case EventType.PAGE:
```

Comment:
Let's support screen calls as well.

Reply:
Actually lemnisk also can have the screen calls but they want us to support only these.